content/blog/botan-first-milestone.md

```diff
@@ -0,0 +1,248 @@
+
+Would it surprise you that they are all directly or indirectly dependent on `crypton`, which contains unaudited C implementations that must be maintained by the Haskell community as a fork after `cryptonite` was abandoned by its original author.
+
+This is a problem, and we're trying to change that with bindings to `botan`.
+
+Having a solid, reliable, and well-maintained cryptography libraries is a huge benefit to the Haskell ecosystem in general, and is essential for most commercial and industrial use cases.
+
+What are you using cryptography for? Let us know in the comments or with upvotes.
+
+## Who are we?
+
+I am Leo Dillinger, a member of [Haskell Cryptography Group][haskell cryptography group], and I am working with the [Haskell Foundation][haskell foundation] to develop free and open-source software for you and the Haskell ecosystem.
+
+## What are our goals?
+
+We seek to provide trusted, open-source cryptography solutions to you. Much of the existing Haskell cryptography ecosystem is aging, unmaintained and unaudited, or very limited in scope, we are seeking to improve that.
+
+Today, we're here to tell you about what is coming next.
+
+## A new phase
+
+As we all know, perfect is the enemy of good; no software is perfect the first time, and we release things when they work, and then continue to improve them. And so we hope that this is simply the first version and first step on a journey of many improvements small and large.
+
+With this milestone, the project enters a new phase in the software development lifecycle - maintenance and development. During initial development, we were nimble, and could make choices arbitrarily - but now that we have something that works, with an initial release and users, we have to keep it working all the while we continue further development. We now have other people invested in this, and can't make choices willy-nilly - we owe it to our users and stakeholders to listen to them.
+
+## The second milestone
+
+That is what this next milestone is about - listening to feedback, improving the user experience, and seeing where the pain points are. Here's what we've heard, and here's what we're planning for the next three months.
+
+### Improved installation support
+
+One of the biggest pieces of feedback that we've received is the need for improved support for the installation of the `botan3` C++ library. This was a recurring item, and we've heard you loud and clear.
+
+We'd like to spend a good chunk of time improving the installation process, with a specific focus on Windows[^1] and Linux support, working on the CI + unit tests for improved reliability and reporting overall.
+
+[^1]: Definitely more highly requested than anticipated
+
+We're also looking into using `build-type: Configure` for bundling Botan C++ as a Haskell package for easy installation on all operating systems - we'd like for usage to be as easy as adding `botan` to your dependencies.
```

Comment:
Wrong idea, especially with [HSEC-2024-002](https://github.com/haskell/security-advisories/pull/157/files#diff-269d148a2ac37251423165ca20885037feef75be318ae44cafd06d7d51f343dcR56-R58) that just came out.

Reply:
Bundled is the wrong term then - pardon my colloquialism; I don't mean actually including source code (I see the obvious flaw in that, and you've warned against it elsewhere) but some way fetching and building the latest version of botan via cabal if it is not already installed. I'll clarify this.

Reply:
Ah I think I see what you mean. Perhaps something like this: https://dashbit.co/blog/rustler-precompiled ?

Reply:
> but some way fetching and building the latest version of botan via cabal if it is not already installed.

This breaks:

- reproducibility (potentially)
- caching and offline installations (you shouldn't need a network connection to install a downloaded cabal package)